kv/kvserver: TestInitRaftGroupOnRequest failed

[(kv/kvserver).TestInitRaftGroupOnRequest failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1884684&tab=buildLog) on [master@ce5d511368b741555344ea28efa40fb9facfa577](https://github.com/cockroachdb/cockroach/commits/ce5d511368b741555344ea28efa40fb9facfa577):

```
=== RUN   TestInitRaftGroupOnRequest
I200419 06:09:30.339213 194256 gossip/gossip.go:404  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42011" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200419 06:09:30.424501 194256 gossip/gossip.go:1524  [n2] no incoming or outgoing connections
I200419 06:09:30.425625 194256 gossip/gossip.go:404  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:45265" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200419 06:09:30.444724 194518 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:42011
I200419 06:09:30.548263 194256 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key /Table/50 [r2] (manual)
I200419 06:09:30.661728 194256 kv/kvserver/replica_command.go:1602  [s1,r2/1:/{Table/50-Max}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r2:/{Table/50-Max} [(n1,s1):1, next=2, gen=1]
I200419 06:09:30.695050 194256 kv/kvserver/replica_raft.go:247  [s1,r2/1:/{Table/50-Max},txn=3340ebf1] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
W200419 06:09:30.706165 194256 kv/kvserver/replica_write.go:200  [s1,r2/1:/{Table/50-Max},txn=3340ebf1] intents not processed as async resolution is disabled
I200419 06:09:30.744250 194256 kv/kvserver/store_snapshot.go:932  [s1,r2/1:/{Table/50-Max}] sending LEARNER snapshot a0600ed1 at applied index 15
I200419 06:09:30.749600 194256 kv/kvserver/store_snapshot.go:975  [s1,r2/1:/{Table/50-Max}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 6, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200419 06:09:30.779800 194632 kv/kvserver/replica_raftstorage.go:780  [s2,r2/2:{-}] applying LEARNER snapshot [id=a0600ed1 index=15]
I200419 06:09:30.794029 194632 kv/kvserver/replica_raftstorage.go:801  [s2,r2/2:/{Table/50-Max}] applied LEARNER snapshot [total=14ms ingestion=4@6ms id=a0600ed1 index=15]
I200419 06:09:30.813318 194664 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r2/1:/{Table/50-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
W200419 06:09:30.815838 194649 kv/kvserver/replica_read.go:102  [liveness-hb,txn=b6485169,range-lookup=/System/NodeLiveness/1,s1,r1/1:/{Min-Table/50}] intents not processed as async resolution is disabled
I200419 06:09:30.819913 194256 kv/kvserver/replica_command.go:1602  [s1,r2/1:/{Table/50-Max}] change replicas (add [(n2,s2):2] remove []): existing descriptor r2:/{Table/50-Max} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=2]
W200419 06:09:30.899312 194653 kv/kvserver/replica_read.go:102  [liveness-hb,txn=3bc0e09c,range-lookup=/System/NodeLiveness/2,s1,r1/1:/{Min-Table/50}] intents not processed as async resolution is disabled
I200419 06:09:30.972088 194256 kv/kvserver/replica_raft.go:247  [s1,r2/1:/{Table/50-Max},txn=721a0ca0] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
W200419 06:09:31.005377 194256 kv/kvserver/replica_write.go:200  [s1,r2/1:/{Table/50-Max},txn=721a0ca0] intents not processed as async resolution is disabled
I200419 06:09:31.006977 194256 util/stop/stopper.go:539  quiescing
I200419 06:09:31.043034 194256 util/stop/stopper.go:539  quiescing
W200419 06:09:31.234785 194733 kv/kvserver/replica_read.go:102  [liveness-hb,txn=40c4ab9a,range-lookup=/System/NodeLiveness/1,s1,r1/1:/{Min-Table/50}] intents not processed as async resolution is disabled
W200419 06:09:31.294020 194940 kv/kvserver/replica_read.go:102  [liveness-hb,txn=2907e88a,range-lookup=/System/NodeLiveness/2,s1,r1/1:/{Min-Table/50}] intents not processed as async resolution is disabled
I200419 06:09:31.410238 194924 util/stop/stopper.go:539  quiescing
I200419 06:09:31.411733 194925 util/stop/stopper.go:539  quiescing
I200419 06:09:31.448956 194923 util/stop/stopper.go:539  quiescing
W200419 06:09:31.452449 194527 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200419 06:09:31.453181 194634 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200419 06:09:31.456630 194606 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200419 06:09:31.457652 194604 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200419 06:09:31.458502 194923 util/stop/stopper.go:539  quiescing
I200419 06:09:31.461017 194923 util/stop/stopper.go:539  quiescing
--- FAIL: TestInitRaftGroupOnRequest (1.24s)
    client_raft_test.go:3976: expected NotLeaseHolderError; got <nil>

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestInitRaftGroupOnRequest PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestInitRaftGroupOnRequest.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestInitRaftGroupOnRequest failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1894666&tab=buildLog) on [master@19b78508bf9f38d473b94d29871e36ba40ea57a0](https://github.com/cockroachdb/cockroach/commits/19b78508bf9f38d473b94d29871e36ba40ea57a0):

```
=== RUN   TestInitRaftGroupOnRequest
I200423 03:31:28.588000 197340 gossip/gossip.go:404  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:44539" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200423 03:31:28.713151 197340 gossip/gossip.go:1524  [n2] no incoming or outgoing connections
I200423 03:31:28.714367 197340 gossip/gossip.go:404  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:42345" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200423 03:31:28.724521 197660 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44539
I200423 03:31:28.770165 197340 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key /Table/50 [r2] (manual)
I200423 03:31:28.888229 197340 kv/kvserver/replica_command.go:1602  [s1,r2/1:/{Table/50-Max}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r2:/{Table/50-Max} [(n1,s1):1, next=2, gen=1]
I200423 03:31:28.922161 197340 kv/kvserver/replica_raft.go:247  [s1,r2/1:/{Table/50-Max},txn=501df2ec] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
W200423 03:31:28.946491 197340 kv/kvserver/replica_write.go:200  [s1,r2/1:/{Table/50-Max},txn=501df2ec] intents not processed as async resolution is disabled
I200423 03:31:28.957758 197340 kv/kvserver/store_snapshot.go:932  [s1,r2/1:/{Table/50-Max}] sending LEARNER snapshot d9ec3ab2 at applied index 15
I200423 03:31:28.959068 197340 kv/kvserver/store_snapshot.go:975  [s1,r2/1:/{Table/50-Max}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 6, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200423 03:31:28.968665 197690 kv/kvserver/replica_raftstorage.go:780  [s2,r2/2:{-}] applying LEARNER snapshot [id=d9ec3ab2 index=15]
I200423 03:31:28.978468 197738 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r2/1:/{Table/50-Max}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200423 03:31:29.007967 197690 kv/kvserver/replica_raftstorage.go:801  [s2,r2/2:/{Table/50-Max}] applied LEARNER snapshot [total=38ms ingestion=4@28ms id=d9ec3ab2 index=15]
I200423 03:31:29.031494 197340 kv/kvserver/replica_command.go:1602  [s1,r2/1:/{Table/50-Max}] change replicas (add [(n2,s2):2] remove []): existing descriptor r2:/{Table/50-Max} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=2]
W200423 03:31:29.078868 197540 kv/kvserver/replica_read.go:102  [liveness-hb,txn=97cb9038,range-lookup=/System/NodeLiveness/1,s1,r1/1:/{Min-Table/50}] intents not processed as async resolution is disabled
I200423 03:31:29.190562 197340 kv/kvserver/replica_raft.go:247  [s1,r2/1:/{Table/50-Max},txn=1e588977] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
W200423 03:31:29.231029 197815 kv/kvserver/replica_read.go:102  [liveness-hb,txn=eb6fa509,range-lookup=/System/NodeLiveness/2,s1,r1/1:/{Min-Table/50}] intents not processed as async resolution is disabled
W200423 03:31:29.236646 197340 kv/kvserver/replica_write.go:200  [s1,r2/1:/{Table/50-Max},txn=1e588977] intents not processed as async resolution is disabled
I200423 03:31:29.243582 197340 util/stop/stopper.go:539  quiescing
W200423 03:31:29.249393 197466 kv/kvserver/store.go:1627  [s1,r1/1:/{Min-Table/50}] could not gossip first range descriptor: node unavailable; try another peer
I200423 03:31:29.282403 197340 util/stop/stopper.go:539  quiescing
W200423 03:31:29.284597 197661 kv/txn.go:603  [liveness-hb] failure aborting transaction: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>; abort caused by: result is ambiguous (server shutdown)
I200423 03:31:29.284918 197661 kv/kvserver/node_liveness.go:832  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200423 03:31:29.496322 197808 kv/kvserver/replica_read.go:102  [liveness-hb,txn=e3f3dbc0,range-lookup=/System/NodeLiveness/1,s1,r1/1:/{Min-Table/50}] intents not processed as async resolution is disabled
W200423 03:31:29.526346 198085 kv/kvserver/replica_read.go:102  [liveness-hb,txn=106c0497,range-lookup=/System/NodeLiveness/2,s1,r1/1:/{Min-Table/50}] intents not processed as async resolution is disabled
I200423 03:31:29.700141 198181 util/stop/stopper.go:539  quiescing
I200423 03:31:29.700290 198182 util/stop/stopper.go:539  quiescing
I200423 03:31:29.754457 198020 util/stop/stopper.go:539  quiescing
W200423 03:31:29.767419 197538 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200423 03:31:29.767550 197739 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200423 03:31:29.768619 197734 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200423 03:31:29.770045 197723 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200423 03:31:29.776934 198020 util/stop/stopper.go:539  quiescing
I200423 03:31:29.778645 198020 util/stop/stopper.go:539  quiescing
--- FAIL: TestInitRaftGroupOnRequest (1.30s)
    client_raft_test.go:3976: expected NotLeaseHolderError; got <nil>

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestInitRaftGroupOnRequest PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestInitRaftGroupOnRequest.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -1803,7 +1803,7 @@ func TestReplicateRestartAfterTruncation(t *testing.T) {
 func runReplicateRestartAfterTruncation(t *testing.T, removeBeforeTruncateAndReAdd bool) {
 	sc := kvserver.TestStoreConfig(nil)
 	// Don't timeout raft leaders or range leases (see the relation between
-	// RaftElectionTimeoutTicks and rangeLeaseActiveDuration). This test expects
+	// RaftElectionTimeoutTicks and RangeLeaseActiveDuration). This test expects
 	// mtc.stores[0] to hold the range lease for range 1.
 	sc.RaftElectionTimeoutTicks = 1000000
 	sc.Clock = nil // manual clock
@@ -3914,6 +3914,12 @@ func TestInitRaftGroupOnRequest(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	storeCfg := kvserver.TestStoreConfig(nil /* clock */)
 	storeCfg.TestingKnobs.DisableMergeQueue = true
+	// Don't timeout range leases (see the relation between
+	// RaftElectionTimeoutTicks and RangeLeaseActiveDuration). This test expects
+	// the replica that holds the lease before the cluster is restarted to
+	// continue holding it after the restart, regardless of how long the restart
+	// takes.
+	storeCfg.RaftElectionTimeoutTicks = 1000000
 	// Disable async intent resolution. This can lead to flakiness in the test
 	// because it allows for the intents written by the split transaction to be
 	// resolved at any time, including after the nodes are restarted. The intent

